protected/commands/FilesCommand.php

```diff
@@ -21,12 +22,86 @@ class FilesCommand extends CConsoleCommand
      */
     public function getHelp()
     {
-        $helpText = "checks files url for a specific dataset in the database" . PHP_EOL;
+        $helpText = "Checks files url for a specific dataset in the database" . PHP_EOL;
         $helpText .= "Usage: ./protected/yiic files checkUrls --doi=<DOI>" . PHP_EOL;
+        $helpText .= "Updates md5 checksum attribute value for all files in a given dataset" . PHP_EOL;
+        $helpText .= "Usage: ./protected/yiic files updateMD5FileAttributes --doi=<DOI>" . PHP_EOL;
 
         return $helpText;
     }
 
+    /**
+     * Update MD5 checksum attribute for all files in a dataset given its DOI
+     * 
+     * @param $doi
+     * @return void
+     */
+    public function actionUpdateMD5FileAttributes($doi) {
+        try {
+            $url = $this->findDatasetMd5FileUrl($doi);
+            echo "Processing $url".PHP_EOL;
+
+            // Dataset id is required for querying files
+            $dataset = Dataset::model()->findByAttributes(array(
+                'identifier' => $doi,
+            ));
+
+            # Download and parse dataset md5 file
+            $contents = DownloadService::downloadFile($url);
+            $lines = explode("\n", $contents);
+            foreach ($lines as $line) {
+                $tokens = explode("  ", $line);
+                // Only parse lines with content in md5 file
+                if($tokens[0] !== "") {
+                    $md5_value = $tokens[0];
+                    $filename = basename($tokens[1]);
+                    if ($filename === "$doi.md5")  // Ignore $doi.md5 file
+                        continue;
+
+                    # Update file_attributes table with md5 checksum value
+                    $file = File::model()->findByAttributes(array(
+                        'dataset_id' => $dataset->id,
+                        'name' => $filename,
+                    ));
+                    $file->updateMd5Checksum($md5_value);
+                }
+            }
+        }
+        catch (Exception $e) {
+            echo $e->getMessage();
+        } 
+        catch (\GuzzleHttp\Exception\GuzzleException $ge) {
+            echo $ge->getMessage();
+        }
+    }
+
+    /**
+     * Returns the URL for a dataset's md5 file
+     *
+     * Determines the URL of the md5 file as it could be:
+     * https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100006/100006.md5
+     * https://ftp.cngb.org/pub/gigadb/pub/10.5524/101001_102000/101001/101001.md5
+     * https://ftp.cngb.org/pub/gigadb/pub/10.5524/102001_103000/102236/102236.md5
+     *
+     * @param $doi
+     * @return string
+     * @throws Exception
+     */
+    private function findDatasetMd5FileUrl($doi): string
+    {
+        // Directory names representing ranges of dataset DOIs
+        $ranges = ['104001_105000', '103001_104000', '102001_103000', '101001_102000', '100001_101000'];
```

Comment:
Hi @pli888,
I feel like it's better to define ``$ranges`` as a constant in the ``Dataset`` class: it's a characteristic about one of ``Dataset``'s property (identifier) and we might want to use it elsewhere in the codebase in the future.

Reply:
A `RANGES` constant that is an array of range directory names has been created in the `Dataset` model class.